File: app/src/main/java/com/example/back2me/api/ItemApiService.java
// file: ItemApiService.java
package com.example.back2me.api;

import retrofit2.Response;
import retrofit2.http.*;
import java.util.List;

public interface ItemApiService {

    @GET("items")
    Response<List<ItemResponse>> getItems();

    @POST("items")
    Response<ItemResponse> createItem(@Body ItemRequest item);

    @GET("items/{id}")
    Response<ItemResponse> getItemById(@Path("id") String id);

    @DELETE("items/{id}")
    Response<Void> deleteItem(@Path("id") String id);
}

File: app/src/main/java/com/example/back2me/api/ItemApiService.kt
// file: ItemApiService.kt (CONFIRMED CORRECT)
package com.example.back2me.api

import retrofit2.Response
import retrofit2.http.*

interface ItemApiService {

    @GET("items")
    suspend fun getItems(): Response<List<ItemResponse>> // CORRECT: No header argument

    @POST("items")
    suspend fun createItem(
        @Body item: ItemRequest
    ): Response<ItemResponse> // CORRECT: No header argument

    @GET("items/{id}")
    suspend fun getItemById(
        @Path("id") id: String
    ): Response<ItemResponse> // CORRECT: No header argument

    @DELETE("items/{id}")
    suspend fun deleteItem(
        @Path("id") id: String
    ): Response<Void> // CORRECT: No header argument
}